rsciio/numpy/_api.py

```diff
@@ -0,0 +1,185 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# RosettaSciIO is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with RosettaSciIO. If not, see <https://www.gnu.org/licenses/#GPL>.
+
+import ast
+import struct
+
+import dask.array as da
+import numpy as np
+
+from rsciio._docstrings import (
+    CHUNKS_DOC,
+    FILENAME_DOC,
+    LAZY_DOC,
+    RETURNS_DOC,
+    SIGNAL_DOC,
+)
+from rsciio.utils.distributed import memmap_distributed
+
+
+def inspect_npy_file(filename):
+    """
+    Inspect a .npy file to extract metadata such as data offset, shape, and dtype.
+
+    Parameters
+    ----------
+    filename : str
+        Path to the .npy file to inspect.
+
+    Returns
+    -------
+    tuple
+        A tuple containing:
+        - data_offset (int): The byte offset where the data starts.
+        - shape (tuple): The shape of the array stored in the file.
+        - dtype (str): The data type of the array elements.
+
+    Example
+    -------
+    >>> offset, shape, dtype = inspect_npy_file('example.npy')
+    >>> print(f"Data offset: {offset} bytes")
+    >>> print(f"Shape: {shape}")
+    >>> print(f"Dtype: {dtype}")
+    """
+    with open(filename, "rb") as f:
+        # Read magic string and version
+        _ = f.read(6)
+        major, _ = struct.unpack("BB", f.read(2))
+
+        encoding = "latin1"  # Default encoding for .npy files version 1.0 and 2.0
+        # Read header length
+        if major == 1:
+            header_len_size = 2
+            header_len = struct.unpack("<H", f.read(header_len_size))[0]
+        elif major in (2, 3):
+            header_len_size = 4
+            header_len = struct.unpack("<I", f.read(header_len_size))[0]
+            if major == 3:
+                encoding = "utf8"  # Version 3.0 uses UTF-8 encoding
+        else:  ## pragma: no cover
+            raise ValueError("Unsupported .npy version")
+
+        header_offset = 6 + 2 + header_len_size
+
+        # Read and parse header
+        header = f.read(header_len).decode(encoding)
+        header_dict = ast.literal_eval(header)
+
+        # Extract metadata
+        shape = header_dict["shape"]
+        dtype = header_dict["descr"]
+        data_offset = header_offset + header_len
+
+        return data_offset, shape, dtype
+
+
+def file_writer(filename, signal, **kwargs):
+    """
+    Write data to npy files.
+
+    Parameters
+    ----------
+    %s
+    %s
+    **kwargs : dict, optional
+        Additional keyword arguments passed to :func:`numpy.save`.
+    """
+    array = signal["data"]
+    if isinstance(array, da.Array):
+        raise TypeError("Lazy signal are not supported for writing to npy files.")
+
```

Comment:
Won't you consider adding a warning here to remind the user that all metadata is lost when saving in this format for the users who don't read the User Guide?

Reply:
I added a note in the docstring. I don't think that we should warn here, because this is expected behaviour: the numpy  doesn't support metadata.
As mentioned above, if we want to go into this consideration, this should be done in a separate PR and done consistently with other plugins.

Reply:
The reason why I think a warning is better is because it is more likely that an user who doesn't know about this limitation reads the warning than the docstring/user guide. However, I may be taking an overcautious appraoch to this. Therefore, let's leave it as is. In the unlikely event that this turns out to be a problem,  somebody will write an issue on it and we'll add the warning then. I'll take the same approach for my other comment.

Reply:
I have open https://github.com/hyperspy/rosettasciio/issues/421 to discuss it.